### PR TITLE
animation fix and improvement

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
### Description of the feature or fix

- fix(anim): fix the callback type error of lv_anim_set_start_cb
- feat(anim): add lv_anim_custom_get which paired with lv_anim_get
- feat(anim): remove time_orig from lv_anim_t

### Checkpoints
- [X] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update [CHANGELOG.md](https://github.com/lvgl/lvgl/blob/master/docs/CHANGELOG.md)
- [ ] Update the documentation

### CODE REVIEW

File: src/misc/lv_anim.c [hunk 1]
```diff
@@ -430,8 +429,10 @@ static void anim_ready_handler(lv_anim_t * a)
             tmp      = a->start_value;
             a->start_value = a->end_value;
             a->end_value   = tmp;
-
-            a->time = a->playback_now == 0 ? a->time_orig : a->playback_time;
+            /*Swap the time and playback_time*/
```
Comment (kisvegabor) [inline src/misc/lv_anim.c]: IMO it's a little bit confusing, if someone gets the animations and checks the fields. The user might ask "How the hell was `time` written to `playback`_time`? Is it a memory corruption? Or did I get the wrong animation? "

Comment (xiaoxiang781216) [reply src/misc/lv_anim.c]: Three reason drive me make this change:

1. The code already change time variable, the user should be surprise first with it.
2. The code already swap start_value and end_value, the change make time/value consistent with each other
3. Could save the four bytes

Another fix is don't change time, but provide a new function:
```
uint32_t lv_anim_get_time(lv_anim_t * a)
{
   a->playback_now ? a->playback_time : a->time_orig;
}
```
And modify all a->time to lv_anim_get_time(a). Actually, this is my first attempt.

Comment (kisvegabor) [reply src/misc/lv_anim.c]: > Three reason drive me make this change:

Sounds reasonable.

> And modify all a->time to lv_anim_get_time(a). Actually, this is my first attempt.

I like it. This way `a->time` is not required and we can rename `a->time_orig` to `a->time`.

Comment (xiaoxiang781216) [reply src/misc/lv_anim.c]: Ok, I will provide new patch in the weekend.

Comment (xiaoxiang781216) [reply src/misc/lv_anim.c]: @kisvegabor I remove time_orig change from PR, please review and merge the rest patch first.

Comment (kisvegabor) [reply src/misc/lv_anim.c]: Ok, great.

**Comment by kisvegabor** [comment]
Thanks!

**Pull request merged**